scheduler/pkg/envoy/processor/server_test.go

```diff
@@ -0,0 +1,156 @@
+	pipelineHandler := pipeline.NewPipelineStore(logger, nil, memoryStore)
+
+	inc := &IncrementalProcessor{
+		cache:            snapCache,
+		logger:           logger,
+		xdsCache:         xdscache.NewSeldonXDSCache(log.New(), &xdscache.PipelineGatewayDetails{}),
+		pipelineHandler:  pipelineHandler,
+		modelStore:       memoryStore,
+		experimentServer: experiment.NewExperimentServer(logger, nil, memoryStore, pipelineHandler),
+		nodeID:           "node_1",
+	}
+
+	err := inc.setListeners()
+	g.Expect(err).To(BeNil())
+
+	conn, err := grpc.NewClient(":18001", grpc.WithTransportCredentials(insecure.NewCredentials()))
+	g.Expect(err).To(BeNil())
+	defer conn.Close()
+
+	c := client.NewADSClient(ctx, &core.Node{Id: "node_1"}, resource.ClusterType)
+	err = c.InitConnect(conn)
+	g.Expect(err).To(BeNil())
+
+	t.Run("Test initial fetch with model version 1", testInitialFetch(g, inc, c))
+	t.Run("Test update model version to 2", testUpdateModelVersion(g, inc, c))
+}
+
+func testInitialFetch(g *WithT, inc *IncrementalProcessor, c client.ADSClient) func(t *testing.T) {
+	expectedClusterNames := []string{"pipelinegateway_http", "pipelinegateway_grpc", "mirror_http", "mirror_grpc", "model_1_grpc", "model_1_http"}
+
+	return func(t *testing.T) {
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+
+		ops := []func(inc *IncrementalProcessor, g *WithT){
+			createTestServer("server", 1),
+			createTestModel("model", "server", 1, []int{0}, 1, []store.ModelReplicaState{store.Available}),
+		}
+
+		for _, op := range ops {
+			op(inc, g)
+		}
+
+		go func() {
+			// watch for configs
+			resp, err := c.Fetch()
+			g.Expect(err).To(BeNil())
+			g.Expect(len(resp.Resources)).To(Equal(6))
+			for _, r := range resp.Resources {
+				cluster := &clusterv3.Cluster{}
+				err := anypb.UnmarshalTo(r, cluster, proto.UnmarshalOptions{})
+				g.Expect(err).To(BeNil())
+				g.Expect(slices.Contains(expectedClusterNames, cluster.Name)).To(BeTrue())
+			}
+
+			err = c.Ack()
+			g.Expect(err).To(BeNil())
+			wg.Done()
+		}()
+
+		wg.Wait()
+	}
+}
+
+func testUpdateModelVersion(g *WithT, inc *IncrementalProcessor, c client.ADSClient) func(t *testing.T) {
+	expectedClusterNames := []string{"pipelinegateway_http", "pipelinegateway_grpc", "mirror_http", "mirror_grpc", "model_2_grpc", "model_2_http"}
+
+	return func(t *testing.T) {
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+
+		ops := []func(inc *IncrementalProcessor, g *WithT){
+			createTestModel("model", "server", 1, []int{0}, 2, []store.ModelReplicaState{store.Available}),
+		}
+
+		for _, op := range ops {
+			op(inc, g)
+		}
+
+		go func() {
+			// watch for configs
+			resp, err := c.Fetch()
+			g.Expect(err).To(BeNil())
+			g.Expect(len(resp.Resources)).To(Equal(6))
+			for _, r := range resp.Resources {
+				cluster := &clusterv3.Cluster{}
+				err := anypb.UnmarshalTo(r, cluster, proto.UnmarshalOptions{})
+				g.Expect(err).To(BeNil())
+				g.Expect(slices.Contains(expectedClusterNames, cluster.Name)).To(BeTrue())
```

Comment:
should we append a list of clusters we get back and then check for equality (instead of `Contains`)?